extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -406,6 +415,83 @@ angular.module('oppia').component('svgFilenameEditor', {
         };
       };
 
+      var loadGroupedObject = function(objId, obj, groupedObjects) {
+        // Checks if the id starts with 'group' to identify whether the
+        // svg objects are grouped together.
+        if (objId.startsWith('group')) {
+          // The objId is of the form "group" + number.
+          var groupId = parseInt(objId.slice(5));
+          // Checks whether the object belongs to an already existing group
+          // or not.
+          if (groupedObjects.length <= groupId) {
+            groupedObjects.push([]);
+          }
+          obj.toSVG = ctrl.createCustomToSVG(
+            obj.toSVG, obj.type, obj.id);
+          groupedObjects[groupId].push(obj);
+        }
+        return groupedObjects;
+      };
+
+      var loadTextObject = function(element, obj) {
+        var childNodes = [].slice.call(element.childNodes);
+        var value = '';
+        var coloredTextIndex = [];
+        // It extracts the text from the tspan tags and appends
+        // with a \n tag to ensure that the texts are subsequent lines.
+        childNodes.forEach(function(el, index) {
+          if (el.nodeName === 'tspan') {
+            value += el.childNodes[0].nodeValue;
+            if (el.style.fill !== '') {
+              // This if block fetches the position of the coloured text so
+              // it can be given color after the text is rendered.
+              coloredTextIndex.push({
+                startIndex: (
+                  value.length - el.childNodes[0].nodeValue.length),
+                endIndex: value.length,
+                fill: el.style.fill,
+                stroke: el.style.stroke,
+                strokeWidth: el.style.strokeWidth
+              });
+            } else if (index < childNodes.length - 1) {
+              value += '\n';
+            }
+          }
+        });
+
+        value = (
+          obj['text-transform'] === 'uppercase' ?
+          value.toUpperCase() : value);
+
+        obj.set({
+          text: value,
+        });
+        var text = new fabric.Textbox(obj.text, obj.toObject());
+        text.set({
+          type: 'textbox',
+          strokeUniform: true,
+        });
+        // The text moves to the right everytime the svg is
+        // rendered so this is to ensure that the text doesn't
+        // render outside the canvas.
```

Comment:
So, does the text still move to the right? If so, is that expected, since usually left is the default alignment, right?

Reply:
Here I am not talking about the alignment, it is about the position of the text. The text itself moves and it is an active issue in fabric js. That is the reason I prevented it from moving outside the canvas.

Reply:
Hi @rohitkatlaa let's link the issue in that case.

Reply:
Done.